### PR TITLE
(feature) Adding auth0 authentication to React

### LINKED ISSUE #20: Integrate auth0 for authentication
- [ ] Configure auth0 service (local, github)

Fontend:
- [ ] Integrate lock into frontend
- [ ] Provide logout functionality
- [ ] Add an interceptor for frontend requests to project service and event system

Project service and event system:
- [x] Configure express-jwt middleware

### PR DESCRIPTION
**Pull request by azangru**
- currently, the app opens the lock if one clicks on the "Sign In" link on the home page
- when email and password are entered, the lock will open a popup window (which will quickly disappear), and, if they are correct, will transition to the Dashboard page
- login with Github still doesn't work

Closes #20

### CODE REVIEW

File: frontend/src/components/app.jsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import React from 'react/addons';
+import {RouteHandler, Navigation} from 'react-router';
+import lockGenerator from '../lock-generator';
+import Home from './home/home.jsx';
+
+let App = React.createClass({
+  mixins: [Navigation],
+
+  componentWillMount() {
+    this.createLock();
```
Comment (wezleytsai) [inline frontend/src/components/app.jsx]: Any reason why you can't just do `this.lock = lockGenerator()` right here?

File: frontend/src/components/home/home.jsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    //     });
+    //   }
+    // }
+    //
+    // this.props.lock.show();
+  },
+
+  render() {
+    return (
+      <div className="homePage">
```
Comment (wezleytsai) [inline frontend/src/components/home/home.jsx]: `className='home-page'`

File: frontend/src/components/app.jsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    this.setState({
+      idToken: localStorage.getItem('userToken')
+    });
+    this.transitionTo('dashboard');
+  },
+
+  render() {
+    if (this.state.idToken) {
+      return (
+        <div className='appContainer'>
```
Comment (wezleytsai) [inline frontend/src/components/app.jsx]: `className='app-container'`, and we should be consistent with the kind of quotes we use for jsx. I think we last agreed on using single quotes like JS.

File: frontend/src/routes.jsx [hunk 4]
```diff
@@ -12,6 +14,7 @@ let DefaultRoute = Router.DefaultRoute;
 let routes = (
   <Route name="app" path="/" handler={App}>
     <DefaultRoute handler={Home} />
+    <Route name="dashboard" handler={Dashboard} />
     <Route name="sprint" handler={SprintBoard} />
   </Route>
 );
```
Comment (wezleytsai) [inline frontend/src/routes.jsx:20]: Use single quotes for lines 15-18, e.g `name='dashboard'`

File: frontend/src/components/app.jsx (hunk 3 above)
Comment (azangru) [inline frontend/src/components/app.jsx]: This again raises the (unimportant) question of whether to render the React app to a div with id="app" like we have here:

https://github.com/the-ninja-turtles/turtle/blob/develop/frontend/server/views/index.ejs#L10

or whether to render it directly into the body. If we render it directly into the body, then the classname of out App component should be `'app'`

**Pull request merged**